### PR TITLE
Added 4 tests for noisy GPs and for GPX

### PR DESCRIPTION
**Pull request by MaelTremouille**
Added 4 tests to complete the tests in the test_ego.py file and added design_space features in GPX

### CODE REVIEW

File: smt/applications/ego.py [hunk 1]
```diff
@@ -71,6 +71,13 @@ def _initialize(self):
             types=int,
             desc="Maximum number of internal optimizations",
         )
+        # New declaration for noise0
```
Comment (relf) [inline smt/applications/ego.py:74]: Remove useless comment (changes are visible in the PR).

File: smt/applications/ego.py [hunk 2]
```diff
@@ -71,6 +71,13 @@ def _initialize(self):
...
+        declare(
+            "noise0",
```
Comment (relf) [inline smt/applications/ego.py:76]: I do not see any usage of options["noise0"]. We do not need to declare a `noise0` option in EGO, it is already an option of `KRG`. To be removed.

File: smt/applications/tests/test_ego.py [hunk 3]
```diff
@@ -87,6 +88,48 @@ def test_function_test_1d(self):
...
+        xlimits = np.array([[0.0, 25.0]])
+        criterion = "EI"
+        design_space = DesignSpace(xlimits)
+        noise0 = [1e-1] # fixed variance for 1D
+
+        ego = EGO(
+            n_iter=n_iter,
+            criterion=criterion,
+            n_doe=3,
+            surrogate=KRG(design_space=design_space, print_global=False),
```
Comment (relf) [inline smt/applications/tests/test_ego.py:123]: `noise0` should be passed in KRG call

File: smt/applications/tests/test_ego.py [hunk 4]
```diff
@@ -87,6 +88,48 @@ def test_function_test_1d(self):
...
+            surrogate=surrogate,
+            random_state=42,
+        )
+
+        x_opt, y_opt, _, _, _ = ego.optimize(fun=TestEGO.function_test_1d)
+
+        self.assertAlmostEqual(18.9, x_opt.item(), delta=1)
+        self.assertAlmostEqual(-15.1, y_opt.item(), delta=1)
+        
+    def test_function_ego_GPN_1D(self):
```
Comment (relf) [inline smt/applications/tests/test_ego.py:112]: maybe use `noisy_KRG` instead of `GPN`

**Review by relf — CHANGES_REQUESTED** [review]
Thanks for your contribution! You should install the VScode extension Ruff and configure to automatically format the code when you save files, otherwise use `ruff format <file>`.

**Comment by MaelTremouille** [comment]
Review of the comments from the last PR and corrections

**Comment by MaelTremouille** [comment]
I formated my files with Ruff

**Review by relf — CHANGES_REQUESTED** [review]
You have to create another branch restarting from master containing the changes related to the feature 'no theta optimization'. Moreover, instead of adding a new option maybe we can use the `hyper_opt` option and add a `NoOptim` value or even better `None` to state that `theta0` is used as is.

**Pull request closed without merging**